docs/pages/guides/react-compiler.mdx

```diff
@@ -169,7 +169,7 @@ module.exports = function (api) {
         {
           'react-compiler': {
             // Passed directly to the React Compiler Babel plugin.
-            compilationMode: 'strict',
+            compilationMode: 'annotation',
```

Comment:
The `strict` value comes from `babel-preset-expo` the plugin has its own settings that it passes to babel-plugin-react-compiler. You can also verify this here in the plugin's source code: https://github.com/expo/expo/blob/db86f82b9b24cdb9eb2849371bdc6ae8dd7d2c44/packages/babel-preset-expo/src/index.ts#L63.

Reply:
I believe there's an issue with the type definition in babel-preset-expo 🤔 

Since these options are passed directly to the React Compiler Babel plugin, passing `'strict'` would be meaningless because the React Compiler doesn't support this value:
https://github.com/facebook/react/blob/4db4b21c63ebc4edc508c5f7674f9df50d8f9744/compiler/packages/babel-plugin-react-compiler/src/Entr
ypoint/Options.ts#L179-L197

Given the current type definition, I think the documentation should use `'infer'` as a valid option instead.

Reply:
And we should consider updating babel-preset-expo's type definition to match the actual React Compiler options: `'infer' |
  'syntax' | 'annotation' | 'all'`.